frontend/packages/db-structure/src/parser/sql/postgresql/converter.ts

```diff
@@ -765,9 +774,17 @@ export const convertToSchema = (
     }
 
     const foreignKeyConstraint = relResult.value
-    const table = tables[foreignTableName]
+    const table = findTable(foreignTableName)
+
     if (table) {
       table.constraints[foreignKeyConstraint.name] = foreignKeyConstraint
+    } else {
+      // Table not found - this could be the cause of missing foreign keys
+      errors.push(
+        new UnexpectedTokenWarningError(
+          `Table "${foreignTableName}" not found when processing foreign key constraint "${foreignKeyConstraint.name}"`,
+        ),
+      )
```

Comment:
@MH4GF 

Is there a possibility that something which previously didn't cause an error might now result in one?
I felt it would be best to prepare a `.changeset` and mark this as at least a minor change.

Reply:
This is a warning, so it just adds a log at runtime. So it is not an error.
However, we need to add a changeset for this behavior!

Reply:
> This is a warning, so it just adds a log at runtime. So it is not an error.

Ah, you're absolutely right! That makes sense! 👍

Reply:
@hoshinotsuyoshi Sorry, I pushed one commit after your review. This warning caused the following deparser test to fail. This is an intentional change as we now need to check for referenced tables.
Could you review it just one more time?

https://github.com/liam-hq/liam/pull/2396/commits/e90911e08f75b3fb433ddc98e5fa058de484d2f6